src/transport/MessageEndpoint.cpp

```diff
@@ -88,171 +95,163 @@ void MessageEndpoint::open(faabric::transport::MessageContext& context,
 void MessageEndpoint::send(uint8_t* serialisedMsg, size_t msgSize, bool more)
 {
     assert(tid == std::this_thread::get_id());
-    assert(this->socket != nullptr);
-
-    if (more) {
-        try {
-            auto res = this->socket->send(zmq::buffer(serialisedMsg, msgSize),
-                                          zmq::send_flags::sndmore);
-            if (res != msgSize) {
-                SPDLOG_ERROR("Sent different bytes than expected (sent "
-                             "{}, expected {})",
-                             res.value_or(0),
-                             msgSize);
-                throw std::runtime_error("Error sending message");
-            }
-        } catch (zmq::error_t& e) {
-            SPDLOG_ERROR("Error sending message: {}", e.what());
-            throw;
-        }
-    } else {
-        try {
-            auto res = this->socket->send(zmq::buffer(serialisedMsg, msgSize),
-                                          zmq::send_flags::none);
-            if (res != msgSize) {
-                SPDLOG_ERROR("Sent different bytes than expected (sent "
-                             "{}, expected {})",
-                             res.value_or(0),
-                             msgSize);
-                throw std::runtime_error("Error sending message");
-            }
-        } catch (zmq::error_t& e) {
-            SPDLOG_ERROR("Error sending message: {}", e.what());
-            throw;
-        }
```

Comment:
This `if`/ `else` was essentially doing the same `send` operation in both branches, just with different `send_flags`, so I merged it into one.